Spec/Auth.swift

```diff
@@ -2091,61 +2091,283 @@ class Auth : QuickSpec {
+                            expect(rest.auth.options.authUrl).toNot(beNil())
+                            expect(rest.auth.options.authParams).toNot(beNil())
+                            expect(rest.auth.options.authHeaders).toNot(beNil())
+                            delay(0.1) { //force to use the authUrl again
+                                done()
+                            }
+                        }
+                    }
+
+                    authOptions.authParams = nil
+                    authOptions.authHeaders = nil
+                    waitUntil(timeout: testTimeout) { done in
+                        rest.auth.authorise(nil, options: authOptions) { tokenDetails, error in
+                            guard let error = error else {
+                                fail("Error is nil"); done(); return
+                            }
+                            expect(error.code).to(equal(400))
+                            expect(tokenDetails).to(beNil())
+                            expect(rest.auth.options.authParams).to(beNil())
+                            expect(rest.auth.options.authHeaders).to(beNil())
+                            done()
+                        }
+                    }
+
+                    // Repeat
+                    waitUntil(timeout: testTimeout) { done in
+                        rest.auth.authorise(nil, options: nil) { tokenDetails, error in
+                            guard let error = error else {
+                                fail("Error is nil"); done(); return
+                            }
+                            expect(error.code).to(equal(400))
+                            expect(tokenDetails).to(beNil())
+                            expect(rest.auth.options.authParams).to(beNil())
+                            expect(rest.auth.options.authHeaders).to(beNil())
+                            done()
+                        }
+                    }
+
+                    authOptions.authUrl = nil
+                    waitUntil(timeout: testTimeout) { done in
+                        rest.auth.authorise(nil, options: authOptions) { tokenDetails, error in
+                            guard let error = error else {
+                                fail("Error is nil"); done(); return
+                            }
+                            expect(UInt(error.code)).to(equal(ARTState.RequestTokenFailed.rawValue))
+                            expect(tokenDetails).to(beNil())
+                            expect(rest.auth.options.authUrl).to(beNil())
+                            expect(rest.auth.options.authParams).to(beNil())
+                            expect(rest.auth.options.authHeaders).to(beNil())
+                            done()
+                        }
+                    }
+
+                    // Repeat
+                    waitUntil(timeout: testTimeout) { done in
+                        rest.auth.authorise(nil, options: nil) { tokenDetails, error in
```

Comment:
Like here, you're passing  `options: nil` and I guess you should do the same in the test above.